project/project.go

```diff
@@ -318,17 +321,16 @@ func (p *State) resolver(runtimeFS *core.Directory, sdk string, gw bkgw.Client,
 
 		st := fsState.Run(
 			llb.Args([]string{entrypointPath}),
-			llb.AddEnv("DAGGER_HOST", "unix:///dagger.sock"),
+			llb.AddEnv("DAGGER_HOST", "unix://"+sessionProxySockPath),
```

Comment:
This is basically temporary until we rework extensions to use the same mechanism as nesting?

Reply:
Yes exactly. I think we should:
1. Put the engine session functionality into shim (want that anyways)
2. Have shim wrap extension execution the same as it does exec

With those two changes, extensions will work the same as exec and this can all be simplified.